File: SocialPulse/SocialPulse.Core/Entities/UserConversation.cs
namespace SocialPulse.Core
{
    public class UserConversation : BaseEntity
    {
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public int ConversationId { get; set; }
        public Conversation Conversation { get; set; } = null!;
    }
}

File: SocialPulse/UI/mobile_socialpulse/lib/models/user_conversation.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mobile_socialpulse/models/user.dart';

import 'conversation.dart';

part 'user_conversation.g.dart';

@JsonSerializable()
class UserConversation{
  int? id;
  int? userId;
  User? user;
  int? conversationId;
  Conversation? conversation;

  UserConversation (this.id,this.userId,this.user,this.conversationId,this.conversation);

  factory UserConversation.fromJson(Map<String, dynamic> json) =>
      _$UserConversationFromJson(json);

  Map<String, dynamic> toJson() => _$UserConversationToJson(this);
}